mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java

```diff
@@ -160,14 +194,62 @@ public void update(CRUDOperationContext ctx,
                 }
                 docIndex++;
             }
+            // if we have any remaining items to update
+            if (numUpdating > 0) {
+                try {
+                    BulkWriteResult result = executeAndLogBulkErrors(bwo);
+                    numUpdated += result.getModifiedCount();
+                } catch (Exception e) {
+                    LOGGER.warn("Update exception for documents for query: {}", query.toString());
+                }
+            }
         } finally {
             if (cursor != null) {
                 cursor.close();
             }
         }
+        handleBulkWriteError(docUpdateErrors, docUpdateAttempts);
+
+        docUpdateActual.stream().filter(doc -> doc != null).forEach(doc -> {
+            ctx.getFactory().getInterceptors().callInterceptors(InterceptPoint.POST_CRUD_UPDATE_DOC, ctx, doc);
+        });
+
         response.setNumUpdated(numUpdated);
         response.setNumFailed(numFailed);
-        response.setNumMatched(docIndex);
+        response.setNumMatched(numMatched);
+    }
+
+    private void handleBulkWriteError(List<BulkWriteError> errors, List<DocCtx> docs) {
+        docUpdateActual.addAll(docs);
+        for (BulkWriteError e : errors) {
```

Comment:
You don't really need docupdateActual, ctx has methods that'll give the the list of docs with no errors.